### PR TITLE
Updates deployment pages in user manual

### PR DESCRIPTION
**Pull request by AlexandreSajus**
- Makes the deployment documentation on Azure, Heroku, and Databricks not directly available
- Removes Windows and Taipy Cloud deployment pages
- Updates the Deployment index page and the mkdocs_template according to available pages

image

image

Resolves https://github.com/Avaiga/taipy-enterprise/issues/406

### CODE REVIEW

File: docs/manuals/run-deploy/deploy/azure/index.md [hunk 1]
```diff
@@ -1,108 +1,7 @@
-[Azure](https://azure.microsoft.com/en-us), one of the world's leading providers of cloud services,
-is a very relevant choice for deploying Taipy applications. It allows developers to stay focused on
-their application bringing effortless scalability, monitoring and security natively. The following
-documentation shows how to deploy on Azure App Service and get an application into the hands of the
-end-user.
+Taipy applications are designed to be easily deployable and customizable. Once deployed, they can natively support multiple users with authentication and authorization. Taipy is cloud-agnostic and can be deployed on any cloud provider or on-premises.
```
Comment (jrobinAV) [inline docs/manuals/run-deploy/deploy/azure/index.md:1]: This sentence does not bring any value for a reader who is searching for how to deploy on Azure.

File: docs/manuals/run-deploy/deploy/azure/index.md [hunk 2]
```diff
@@ -1,108 +1,7 @@
...
 
-## Prerequisites
+!!! note "Taipy Enterprise edition"
 
-- Minimal knowledge of Azure.
-- Azure CLI should be installed. Check
-  [the official documentation](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli) for
-  Azure CLI installation.
-- [:material-arrow-right: Running a Taipy application](../../run/index.md)
+    This part of the documentation is available on demand. Please contact us using the link below. Taipy Deployment Support, Scripts and Documentation are readily available in the Enterprise edition.
```
Comment (jrobinAV) [inline docs/manuals/run-deploy/deploy/azure/index.md:5]: This is not what we said.
1. The doc is available on demand, and not only for Enterprise Edition customers.
2. The Enterprise Edition customers benefit from 2 days free of charge of consulting and support to deploy user applications with them.

File: docs/tutorials/large_data/3_decimator/index.md
Comment (jrobinAV) [inline docs/tutorials/large_data/3_decimator/index.md:1]: This is unrelated. Please rollback and create another PR. Thx

**Review by jrobinAV — COMMENTED** [review]
Please break lines after 100 characters.

**Comment by FlorianJacta** [comment]
Agree with JR and merge develop to your branch to resolve conflicts.

File: docs/manuals/run-deploy/deploy/azure/index.md (hunk 1 above)
Comment (AlexandreSajus) [reply docs/manuals/run-deploy/deploy/azure/index.md:1]: What should I write then? Only the disclaimer like so?
image

File: docs/manuals/run-deploy/deploy/azure/index.md (hunk 2 above)
Comment (AlexandreSajus) [reply docs/manuals/run-deploy/deploy/azure/index.md:5]: My sentence seems to fit 1. For 2. I don't think it is wise to go into details in what we provide in an Enterprise package without talking with the lead first. What sentence do you propose?

File: docs/tutorials/large_data/3_decimator/index.md
Comment (AlexandreSajus) [reply docs/tutorials/large_data/3_decimator/index.md:1]: Ah my bad, I'm not sure what happened. I'll wait for your feedback on the other comments and then rollback and PR

File: docs/manuals/run-deploy/deploy/azure/index.md (hunk 2 above)
Comment (jrobinAV) [reply docs/manuals/run-deploy/deploy/azure/index.md:5]: That's what was discussed yesterday.

File: docs/manuals/run-deploy/deploy/azure/index.md (hunk 1 above)
Comment (jrobinAV) [reply docs/manuals/run-deploy/deploy/azure/index.md:1]: Something like the following:

``` suggestion
The documentation page is available on demand. Please contact us to get it.
[Contact us](https://taipy.io/book-a-call){: .tp-btn .tp-btn--accent target='blank' }

!!! "Taipy Enterprise Edition"

    Besides this documentation page, Enterprise Edition customers benefit from 
    free-of-charge consulting and support to deploy user applications. Check out the
    [Taipy Enterprise Edition](http://toto.tata.com).
```

Don't forget the capital E to Edition.

Comment (AlexandreSajus) [reply docs/manuals/run-deploy/deploy/azure/index.md:1]: Understood! I'll do that

File: docs/manuals/run-deploy/deploy/azure/index.md (hunk 2 above)
Comment (AlexandreSajus) [reply docs/manuals/run-deploy/deploy/azure/index.md:5]: Let me ask Des' input on this before going forward with this

Comment (AlexandreSajus) [reply docs/manuals/run-deploy/deploy/azure/index.md:5]: Vincent agrees with the idea of not precising what the number of days of support we provide in the Enterprise license. I'll go ahead and create a new PR for the other changes if that's okay with you.

Comment (jrobinAV) [reply docs/manuals/run-deploy/deploy/azure/index.md:5]: Yep
My proposal was this: 
    Besides this documentation page, Enterprise Edition customers benefit from 
    free-of-charge consulting and support to deploy user applications. Check out the
    [Taipy Enterprise Edition](http://toto.tata.com).

Comment (AlexandreSajus) [reply docs/manuals/run-deploy/deploy/azure/index.md:5]: Yes, this is good, I'll open a new PR with this

**Comment by AlexandreSajus** [comment]
Done in #1045
